www/.vuepress/components/ExternalLibIconsShowcase.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="card-gallery">
+     <div class="card" @click="copyToClipboard('pie-chart')">
+      <fw-icon name="pie-chart" size="20" library="feather" ></fw-icon>
+      <span class="name">pie-chart</span>
+      <span id='pie-chart' class="copied">-Copied- library: feather</span>
+    </div>
+    <div class="card" @click="copyToClipboard('cloud')">
+      <fw-icon name="cloud" size="20" library="heroicons" ></fw-icon>
+      <span class="name">cloud</span>
+      <span id='cloud' class="copied">-Copied- library: heroicons</span>
+    </div>
+    <div class="card" @click="copyToClipboard('Header')">
+      <fw-icon name="Header" size="20" library="crayons" ></fw-icon>
+      <span class="name">Header</span>
+      <span id='Header' class="copied">-Copied- library: crayons</span>
+    </div> 
+     <div class="card" @click="copyToClipboard('calendar')">
+      <fw-icon name="calendar" size="20" library="system" ></fw-icon>
+      <span class="name">calendar</span>
+      <span id='calendar' class="copied">-Copied- library: system</span>
+    </div>    
+  </div>
+</template>
+
+<script>
+import IconLibRegistry from '@freshworks/crayons-icon/icon-utils/icon-registry';
+export default {
+  name: 'icon-external-gallery',
+  data() {
+    return {
+      icons: [],
+    }
+  },
+  methods: {
+
+    async copyToClipboard(text) {
+      const el = document.createElement('textarea');
+      el.value = text;
+      document.body.appendChild(el);
+      el.select();
+      document.execCommand('copy');
+      document.body.removeChild(el);
+      document.getElementById(text).style.display='block';
+      setTimeout(() => document.getElementById(text).style.display='none', 2000);
+    },
+  },
+  mounted() {
+    IconLibRegistry.registerIconLib([
+                                      {lib:'feather', path: 'https://cdn.jsdelivr.net/npm/feather-icons@4.28.0/dist/icons'},
+                                      {lib:'heroicons', path: 'https://cdn.jsdelivr.net/npm/heroicons@0.4.2/outline'},
+                                      {lib:'system', path: 'http://127.0.0.1:8081/custom-icons'},
+                                      {lib:'crayons', path: 'https://unpkg.com/@freshworks/crayons@3.0.0-beta.4/dist/crayons/icon-assets/icons'}
+                                    ]);
+    const svgDocEle = '{"svg":{"props":{"viewBox":"0 0 615 84","style":"margin-top:30px;"}},"g":null, "paths":[{"props":null},{"props":{"fill":"black"}}]}';
+    try{
+       IconLibRegistry.mutateSVG('system', 'crayons-logo', svgDocEle);
+    }catch(ex){
+      //console.error(ex);
```

Comment:
You can uncomment it or add a proper log message here

Reply:
closed